provider/src/tzif.rs

```diff
@@ -0,0 +1,142 @@
+#[databake(path = temporal_provider::tzif)]
+pub struct ZoneInfoProvider<'data> {
+    // IANA identifier map to TZif index.
+    pub ids: ZeroAsciiIgnoreCaseTrie<ZeroVec<'data, u8>>,
+    // Vector of TZif data
+    pub tzifs: VarZeroVec<'data, ZeroTzifULE, Index32>,
+}
+
+#[zerovec::make_varule(ZeroTzifULE)]
+#[derive(PartialEq, Debug, Clone, yoke::Yokeable, serde::Serialize, databake::Bake)]
+#[zerovec::skip_derive(Ord)]
+#[zerovec::derive(Debug, Serialize)]
+#[databake(path = temporal_provider::tzif)]
+pub struct ZeroTzif<'data> {
+    pub transitions: ZeroVec<'data, i64>,
+    pub transition_types: ZeroVec<'data, u8>,
+    // NOTE: zoneinfo64 does a fun little bitmap str
+    pub types: ZeroVec<'data, LocalTimeRecord>,
+    pub posix: Cow<'data, str>,
+}
+
+#[zerovec::make_ule(LocalTimeRecordULE)]
+#[derive(
+    PartialEq,
+    Eq,
+    Debug,
+    Clone,
+    Copy,
+    PartialOrd,
+    Ord,
+    yoke::Yokeable,
+    serde::Serialize,
+    databake::Bake,
+)]
+#[databake(path = temporal_provider::tzif)]
+pub struct LocalTimeRecord {
+    pub offset: i64,
+    pub is_dst: bool,
+}
+
+impl From<&zoneinfo_compiler::tzif::LocalTimeRecord> for LocalTimeRecord {
+    fn from(value: &zoneinfo_compiler::tzif::LocalTimeRecord) -> Self {
+        Self {
+            offset: value.offset,
+            is_dst: value.is_dst,
+        }
+    }
+}
+
+impl ZeroTzif<'_> {
+    fn from_transition_data(data: &ZoneInfoTransitionData) -> Self {
+        let tzif = data.to_v2_data_block();
+        let transitions = ZeroVec::alloc_from_slice(&tzif.transition_times);
+        let transition_types = ZeroVec::alloc_from_slice(&tzif.transition_types);
+        let mapped_local_records: Vec<LocalTimeRecord> =
+            tzif.local_time_types.iter().map(Into::into).collect();
+        let types = ZeroVec::alloc_from_slice(&mapped_local_records);
+        let posix = String::from("TODO").into();
+
+        Self {
+            transitions,
+            transition_types,
+            types,
+            posix,
+        }
+    }
+}
+
+#[derive(Debug)]
+pub enum ZoneInfoDataError {
+    Build(ZeroTrieBuildError),
+}
+
+impl ZoneInfoProvider<'_> {
+    pub fn build(tzdata: &Path) -> Result<Self, ZoneInfoDataError> {
+        let mut provider = TzdbDataProvider::try_from_zoneinfo_directory(tzdata).unwrap();
+        let mut identifiers = BTreeMap::default();
+        let mut zones_set = BTreeSet::default();
+
+        let zoneinfo_compiled = provider.zone_info.associate_and_build();
+
+        for zone_identifier in provider.zone_info.zones.keys() {
+            let _ = zones_set.insert(zone_identifier.clone());
+            identifiers.insert(zone_identifier.clone(), zone_identifier.clone());
+        }
+        for (link, zone) in provider.zone_info.links.iter() {
+            identifiers.insert(link.clone(), zone.clone());
+        }
+
+        let zones: Vec<String> = zones_set.iter().cloned().collect();
```

Comment:
suggestion: It's not necessary to use a set struct and then switch to `Vec` later; since the set is built from unique map keys, its entries are guaranteed to be unique anyhow.

Reply:
Hmmmm, yeah, that's a good point.